utils/sccli.py

```diff
@@ -55,6 +56,36 @@ def system(cmd):
     returncode = ret.returncode
     return out, err, returncode
 
+def set_proxy():
+    proxy = os.getenv('PROXY', '')
+    if proxy:
+        proxy_user = quote_plus(os.getenv('PROXY_USER',''))
+        if proxy_user:
+            proxy_password = quote_plus(os.getenv('PROXY_PASSWORD',''))
+            http_proxy_url = "http://%s:%s@%s" % (proxy_user, proxy_password, proxy)
+            https_proxy_url = "https://%s:%s@%s" % (proxy_user, proxy_password, proxy)
+        else:
+            http_proxy_url = "http://%s" % proxy
+            https_proxy_url = "https://%s" % proxy
+        # openshift proxy setup
+        if system(('sed -i -e "/^#HTTP_PROXY=*/cHTTP_PROXY=%s"'
+                   ' -e "/^#HTTPS_PROXY=*/cHTTPS_PROXY=%s"'
+                ' %s') % (http_proxy_url, https_proxy_url, OPENSHIFT_OPTION))[2]:
+            return ("Permisison denined: %s" % OPENSHIFT_OPTION)
+        # docker daemon proxy setup
+        if not os.path.isdir('/etc/systemd/system/docker.service.d'):
+            subprocess.call("mkdir /etc/systemd/system/docker.service.d", shell=True)
+        env_file_content = ('[Service]\n'
+                'Environment="HTTP_PROXY=%s" "HTTPS_PROXY=%s" "NO_PROXY=localhost,127.0.0.0/8,::1"\n') \
```

Comment:
Docker doesn't accept CIDR notation in `NO_PROXY`, and this needs to cover the registry... see openshift/openshift-ansible#1919 for more details.